modules/exploits/windows/local/ms10_092_schelevator.rb

```diff
@@ -60,77 +85,64 @@ def initialize(info = {})
     )
 
     register_options([
-      OptString.new("CMD", [ false, "Command to execute instead of a payload" ]),
-      OptString.new("TASKNAME", [ false, "A name for the created task (default random)" ]),
+      OptString.new('TASKNAME', [ false, 'A name for the created task (default random)' ]),
     ])
   end
 
   def check
-    vuln = false
-    winver = sysinfo["OS"]
-    affected = [ 'Windows Vista', 'Windows 7', 'Windows 2008' ]
-    affected.each { |v|
-      if winver.include? v
-        vuln = true
-        break
-      end
-    }
-    if not vuln
-      return Exploit::CheckCode::Safe
+    return CheckCode::Safe("Target OS #{session.platform} is not vulnerable") unless session.platform == 'windows'
+
+    system_info = sysinfo
+    return CheckCode::Unknown('Could not retrieve OS system information.') unless system_info
+
+    [ 'Windows Vista', 'Windows 7', 'Windows 2008' ].each do |v|
+      return CheckCode::Detected if system_info['OS'].include?(v)
     end
 
-    return Exploit::CheckCode::Appears
+    CheckCode::Safe("#{system_info['OS']} is not vulnerable")
   end
 
   def exploit
-    if sysinfo["Architecture"] == ARCH_X64 && session.arch == ARCH_X86
-      #
-      # WOW64 Filesystem Redirection prevents us opening the file directly. To make matters
-      # worse, meterpreter/railgun creates things in a new thread, making it much more
-      # difficult to disable via Wow64EnableWow64FsRedirection. Until we can get around this,
-      # offer a workaround and error out.
-      #
-      fail_with(Failure::NoTarget, "Running against via WOW64 is not supported, try using an x64 meterpreter...")
-    end
+    fail_with(Failure::None, 'Session is already elevated') if is_system?
+    fail_with(Failure::NoTarget, 'This exploit requires a native Windows meterpreter session') unless session.type == 'meterpreter'
```

Comment:
The Python and other meterpreters would pass this check. It looks like this exploit should work with more Meterpreters than just the Windows meterpreter.

```suggestion
    fail_with(Failure::NoTarget, 'This exploit requires a meterpreter session') unless session.type == 'meterpreter'
```

Reply:
done